pkg/haproxy/haproxy.go

```diff
@@ -291,6 +291,27 @@ func haproxyContainer(cr *apiv1alpha1.PerconaServerMySQL) corev1.Container {
 	}
 	env = append(env, spec.Env...)
 
+	readinessProbe := spec.ReadinessProbe
+	readinessProbe.Exec = &corev1.ExecAction{
+		Command: []string{"/opt/percona/haproxy_readiness_check.sh"},
+	}
+	livenessProbe := spec.LivenessProbe
+	livenessProbe.Exec = &corev1.ExecAction{
+		Command: []string{"/opt/percona/haproxy_liveness_check.sh"},
+	}
+
+	probsEnvs := []corev1.EnvVar{
+		{
+			Name:  "LIVENESS_CHECK_TIMEOUT",
+			Value: fmt.Sprint(livenessProbe.TimeoutSeconds),
+		},
+		{
+			Name:  "READINESS_CHECK_TIMEOUT",
+			Value: fmt.Sprint(readinessProbe.TimeoutSeconds),
+		},
+	}
+	env = append(env, probsEnvs...)
+
```

Comment:
We can add a test case for the probes in `haproxy_test.go` checking that the probe handlers have the right exec command, that the various seconds configs exist and also that the env vars used by the probe bash scripts are set up properly.